dhis-2/dhis-services/dhis-service-core/src/main/resources/org/hisp/dhis/category/hibernate/Category.hbm.xml

```diff
@@ -39,13 +39,12 @@
                     foreign-key="fk_category_categoryoptionid"/>
     </list>
 
-    <list name="categoryCombos" table="categorycombos_categories" inverse="true">
+    <bag name="categoryCombos" table="categorycombos_categories" inverse="true">
```

Comment:
Just curious, are we sure we want to use the more exotic `bag` collection type? Do we use it other places? Feels a bit arbitrary if we only use it here. Are we handling sort order gaps consistently other places, as this might happen to all lists I assume?

Reply:
This is a _hibernate mapping error_ of `Category` and `CategoryCombo` association. It's different from the `OptionSet.opion` or any other `<list>` sortOrder issue that i have fixed before. 
In details:
- We have a many-to-many association between `Category` and `CategoryCombo` and we wanted `list-index` to be implemented on both sides. Means that  both `Category.categoryCombos`  and `CategoryCombo.categories` have `list-index`. 
- However the issue is that in table `categorycombos_categories` **there is only one `sort_oder` column used by two `<list-index>`**. Hence, whatever sortOrders are saved correctly for one side of this association should be incorrect in the other side.
- Now, we can fix this by adding one more `sortOrder` column to that table. But if you have a look at Maintenance App > Category edit page, there is no `CategoryCombo` list. Which means the `Category.categoryCombos` sortOrder is not in used. I guess it has been used by a Struts App long time ago. Therefore instead of fixing a feature that is not in used, I think it's better to remove it.
- Why this issue only happen from 2.39 ? The reason is from 2.39 we moved from our custom `NodeDeserialiser` to `JacksonDeserializer`.  The `NodeDeserialiser` may have excluded the `null` items from the collection when converting the list of Objects to `org.hisp.dhis.node.Node`.
- After this, we can create a jira issue and go over all of remaining `<list>` mapping to verify if the `sortOrder` is in used. If not we can convert them to `<bag>` to avoid  unintended issues. 
- To expand the issue further, currently there are two ways of mapping `list-index` in our codes.
1. **Implicit**: there is no `sortOrder` property in Java Object model. So the order values are implicitly generated and saved by hibernate base on the Java List indexes.
2. **Explicit**: there is a `sortOrder` property in Java Object model. User manually set the order value to the object and hibernate will save it to database. 
Until now, approach (1) has not caused any error beside the issue fixed by this PR, but as mentioned, it's a many-to-many mapping error. While approach (2) has caused many issues for `OptionSet.option`.